added nginx ECHO module option

ultrafunkamsterdam opened the pull request:
Thanks for this nice script, which i'm actually using quite a lot. 
I just added the echo module for my own usage. Here is the pr in case you are interested.

ultrafunkamsterdam committed:
added nginx ECHO module option

angristan commented:
Thanks, please format the script with `shfmt -s -w nginx-autoinstall.sh` and add the module to the README list 🙏

ultrafunkamsterdam committed:
shfmt -s -w nginx-autoinstall.sh

ultrafunkamsterdam committed:
Merge branch 'angristan:master' into master

xopez committed:
Update nginx mainline version to 1.21.2

xopez committed:
Update OpenSSL to 1.1.1l (#202)

xopez committed:
Update OpenSSL to 1.1.1l (#202)

karelbasta committed:
Add patch for NGINX 1.19.7+ and QUIC (#192)

karelbasta committed:
Add patch for NGINX 1.19.7+ and QUIC (#192)

MaximeMichaud committed:
NGINX_MAINLINE 1.21.2 ==> 1.21.4 (#210)

MaximeMichaud committed:
NGINX_MAINLINE 1.21.2 ==> 1.21.4 (#210)

angristan committed:
Remove tests on Debian 8, add Debian 11

angristan committed:
Remove tests on Debian 8, add Debian 11

MaximeMichaud committed:
NGINX_MAINLINE 1.21.4 ==> 1.21.5 (#217)

MaximeMichaud committed:
NGINX_MAINLINE 1.21.4 ==> 1.21.5 (#217)

MaximeMichaud committed:
NGINX_MAINLINE 1.21.5 ==> 1.21.6 (#219)

MaximeMichaud committed:
NGINX_MAINLINE 1.21.5 ==> 1.21.6 (#219)

Commit:
Fix quiche path

Since the quiche repository refactor in commit [#bbe6e2a](https://github.com/cloudflare/quiche/commit/bbe6e2ab22126773bb63e4605ac4c8e7594f002a) path for boringssl needs an update

Commit:
Fix quiche path

Since the quiche repository refactor in commit [#bbe6e2a](https://github.com/cloudflare/quiche/commit/bbe6e2ab22126773bb63e4605ac4c8e7594f002a) path for boringssl needs an update

Commit:
ci: enable HTTP/3 Quiche patch

I tried to recreate your [#f0f8714](https://github.com/angristan/nginx-autoinstall/commit/f0f8714b50a23b49da5d8efe5b77262494d81609) just the opposite way.

Commit:
ci: enable HTTP/3 Quiche patch

I tried to recreate your [#f0f8714](https://github.com/angristan/nginx-autoinstall/commit/f0f8714b50a23b49da5d8efe5b77262494d81609) just the opposite way.

Commit:
Fix nginx-1.16 path

Commit:
Fix nginx-1.16 path

droplinxuser committed:
Remove duplicate RTMP

Remove duplicate "RTMP=${RTMP:-n}".

droplinxuser committed:
Remove duplicate RTMP

Remove duplicate "RTMP=${RTMP:-n}".

dependabot[bot] committed:
Bump actions/checkout from 2 to 3

Bumps [actions/checkout](https://github.com/actions/checkout) from 2 to 3.
- [Release notes](https://github.com/actions/checkout/releases)
- [Changelog](https://github.com/actions/checkout/blob/main/CHANGELOG.md)
- [Commits](https://github.com/actions/checkout/compare/v2...v3)

---
updated-dependencies:
- dependency-name: actions/checkout
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump actions/checkout from 2 to 3

Bumps [actions/checkout](https://github.com/actions/checkout) from 2 to 3.
- [Release notes](https://github.com/actions/checkout/releases)
- [Changelog](https://github.com/actions/checkout/blob/main/CHANGELOG.md)
- [Commits](https://github.com/actions/checkout/compare/v2...v3)

tmiland committed:
Add cache modules

Added cache modules.

tmiland committed:
Remove duplicate subfilter

Removed duplicate subfilter.

tmiland committed:
Reconstruct setmisc

Reconstructured setmisc.

tmiland committed:
Add echo-nginx-module

Added echo-nginx-module.

tmiland committed:
Fix echo-nginx-module tarball path

Fixed  echo-nginx-module tarball path.

tmiland committed:
Update README.md

tmiland committed:
Update README.md

tmiland committed:
Update wrapper.sh for Cache modules

- Updated wrapper.sh for Cache modules
- Removed duplicates (RTMP,SUBFILTER)

tmiland committed:
Update wrapper.sh for Cache modules

- Updated wrapper.sh for Cache modules
- Removed duplicates (RTMP,SUBFILTER)

tmiland committed:
Add missing space

tmiland committed:
Add missing space

tmiland committed:
Add lua-resty-core & lua-resty-lrucache

- Added lua-resty-core & lua-resty-lrucache.
- Updated lua-nginx & luajit2

tmiland committed:
Update wrapper.sh

tmiland committed:
Update wrapper.sh

tmiland committed:
Add license authentication for GeoLite2

- Added license authentication to download GeoLite2.
- Added ASN database
- added crontab for geoipupdate
  - This adds GeoIP.conf to /usr/local/etc with credentials
- Added change main to main contrib in /etc/apt/sources.list (to install geoipupdate from APT)
- Added credentials to headless install

tmiland committed:
Switch repo for GeoIP.conf

- Switched repo for GeoIP.conf

tmiland committed:
Move options and modules to headless

- Moved options and modules to headless with default options/modules.

tmiland committed:
Move default versions/options & modules out of headless=y

Moved default versions/options & modules out of headless=y to fix non-headless install.

MaximeMichaud committed:
nginx-1.22.0 stable version has been released

nginx-1.22.0 stable version has been released, incorporating new features and bug fixes from the 1.21.x mainline branch  — including hardening against potential requests smuggling and cross-protocol attacks, ALPN support in the stream module, better distribution of connections among worker processes on Linux, support for the PCRE2 library, support for OpenSSL 3.0 and SSL_sendfile(), improved sendfile handling on FreeBSD, the mp4_start_key_frame directive, and more.

HEADLESS=y \
NGINX_VER=STABLE \
BROTLI=y \
HEADERMOD=y \
MODSEC=n \
MODSEC_ENABLE=n \
bash /root/nginx-autoinstall.sh

nginx version: nginx/1.22.0
built by gcc 10.2.1 20210110 (Debian 10.2.1-6) 
built with OpenSSL 1.1.1n  15 Mar 2022
TLS SNI support enabled
configure arguments: --prefix=/etc/nginx --sbin-path=/usr/sbin/nginx --conf-path=/etc/nginx/nginx.conf --error-log-path=/var/log/nginx/error.log --http-log-path=/var/log/nginx/access.log --pid-path=/var/run/nginx.pid --lock-path=/var/run/nginx.lock --http-client-body-temp-path=/var/cache/nginx/client_temp --http-proxy-temp-path=/var/cache/nginx/proxy_temp --http-fastcgi-temp-path=/var/cache/nginx/fastcgi_temp --user=nginx --group=nginx --with-cc-opt=-Wno-deprecated-declarations --with-cc-opt=-Wno-ignored-qualifiers --with-threads --with-file-aio --with-http_ssl_module --with-http_v2_module --with-http_mp4_module --with-http_auth_request_module --with-http_slice_module --with-http_stub_status_module --with-http_realip_module --with-http_sub_module --add-module=/usr/local/src/nginx/modules/ngx_brotli --add-module=/usr/local/src/nginx/modules/headers-more-nginx-module-0.33

Commit:
merge

Commit:
fix

Commit:
merge
